### ISSUE TITLE
Does not OOM and stop

### ISSUE THREAD

**Issue by vaibhav-kaushal** [seznam/swift-uniredis#3]
I basically ran a loop (macOS) that looked like this:

```swift
import Foundation
import UniRedis

let redis = try UniRedis("redis://localhost:6379")

while i < 10_000_000 {
	print(i)
	do {
		let c = try redis.cmd("ZADD", params: [ "somekey", "\(i)", "abcdefghijklmnopqrstuvwxyz\(i)"])
	} catch {
		print("Error")
		exit(0)
	}
	i += 1
}
```

The loop was supposed to go over the `maxmemory` limit I had set in the redis config file. However that did not happen. The loop went on and printed the last `i` value at `1604848` before I pressed Ctrl+C. However, when I tried to get the number of items in the set, I got this: 

```
127.0.0.1:6379> ZCARD somekey
(integer) 1377312
```

So the code was supposed to crash and print `Error` and die but did not. What is the problem here?

**Comment by danielfojt** [comment]
Thanks for reporting this.

In case the response from redis is valid, UniRedis.cmd() currently just returns the response to caller. So it's actually a missing check in the example code above, like:

```swift
guard c.type != .error else {
     let err = c.content as? String
     throw UniRedisError.error(detail: err!)
}
```

But I see, it's quite impractical, and it'd be better to throw exception earlier from .cmd() call.

Would you please test the updated code from branch [resp-err-throw](https://github.com/seznam/swift-uniredis/compare/resp-err-throw)? Your example code should work properly with this change, ie. exception should be caught and the program should exit(0).

**Comment by vaibhav-kaushal** [comment]
@danielfojt I am pretty new to swift and I just wrote a small program with UniRedis as a dependency. I am not sure how to use a custom branch as a dependency. 😕

**Comment by danielfojt** [comment]
OK, nevermind. So please just test your code with 'guard' statement mentioned above, placed right after your redis.cmd() call. It's basically the same change I've made in "resp-err-throw" branch. Thank you.

**Comment by danielfojt** [comment]
I've pushed commit 5854551+ to master, and extended tests to cover the case, commit 590792e+. So now it's fixed in 0.14.0.

Thank you for reporting this.
